controllers/discovery/initDiscoveryUserData.js

```diff
@@ -20,6 +21,14 @@ const InitDiscoveryUserData = async (req, res) => {
     where_anon_dm = 'AND A.allow_anon_dm = :allow_anon_dm';
   }
 
+  const blockedIds = await UsersFunction.getBlockedAndBlockerUserId(UserBlockedUser, req.userId);
+  let filterBlockedUser = '';
+  if (blockedIds.length > 0) {
+    filterBlockedUser += `AND A.user_id NOT IN (${blockedIds
+      .map((item) => `'${item}'`)
+      .join(',')})`;
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code is constructing a SQL query string using JavaScript template literals. This can potentially lead to SQL injection if the `blockedIds` are not properly sanitized. Although it seems like `UsersFunction.getBlockedAndBlockerUserId` should return an array of IDs (which should be safe), it's better to use parameterized queries or prepared statements for security reasons.

```diff
- filterBlockedUser += `AND A.user_id NOT IN (${blockedIds
-   .map((item) => `'${item}'`)
-   .join(',')})`;
+ filterBlockedUser += 'AND A.user_id NOT IN (:blockedIds)';
```

Then, when executing the query, pass `blockedIds` as a parameter:

```diff
- const result = await sequelize.query(totalDataQuery, { replacements: { ... }, type: QueryTypes.SELECT });
+ const result = await sequelize.query(totalDataQuery, { replacements: { ..., blockedIds }, type: QueryTypes.SELECT });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->